tests/ConnectionTest.php

```diff
@@ -396,6 +397,48 @@ public function testConnectStartsTransactionInNoAutoCommitMode(): void
         self::assertTrue($conn->isTransactionActive());
     }
 
+    public function testTransactionIsNotActiveAfterTransactionalInAutoCommitMode(): void
+    {
+        $driverMock = $this->createMock(Driver::class);
+        $driverMock->expects(self::any())
```

Comment:
I don't think `->expects(self::any())` achieves anything, it can be removed IMO

```suggestion
        $driverMock
```

There are other occurrences below.

Reply:
> Required parameter '$invocationRule' missing